### PR TITLE
fix(levm): codecopy: return a 0 array if the offset is larger than the bytecode length. 

### PR DESCRIPTION
**Pull request by lima-limon-inc**
**Motivation**
Currently, the offset is cast to usize immediately . This causes an a `VeryLargeNumber` number error, and thus the memory is never updated.

**Description**
Now, we check if the offset is larger than the bytecode itself. If that is the case, then we simply return an array of `size` many zeros. 

This won't compile because it depends on the changes made here: https://github.com/lambdaclass/ethrex/pull/1458

### CODE REVIEW

**Comment by lima-limon-inc** [comment]
Note: I merged Lean's commit before it got merged into main to test this commit out.

File: crates/vm/levm/src/opcode_handlers/environment.rs [hunk 1]
```diff
@@ -243,15 +231,21 @@ impl VM {
...
+            for (i, byte) in current_call_frame
+                .bytecode
+                .iter()
+                .skip(code_offset)
+                .take(size)
+                .enumerate()
+            {
+                if let Some(data_byte) = data.get_mut(i) {
+                    *data_byte = *byte;
+                }
```
Comment (lima-limon-inc) [inline crates/vm/levm/src/opcode_handlers/environment.rs:248]: This is my change

File: crates/vm/levm/src/opcode_handlers/environment.rs [hunk 2]
```diff
@@ -215,16 +211,8 @@ impl VM {
...
-            .pop()?
-            .try_into()
-            .map_err(|_| VMError::VeryLargeNumber)?;
-        let code_offset: usize = current_call_frame
-            .stack
-            .pop()?
-            .try_into()
-            .map_err(|_| VMError::VeryLargeNumber)?;
+        let destination_offset = current_call_frame.stack.pop()?;
+        let code_offset = current_call_frame.stack.pop()?;
```
Comment (lima-limon-inc) [inline crates/vm/levm/src/opcode_handlers/environment.rs:215]: Ditto

File: crates/vm/levm/src/opcode_handlers/environment.rs [hunk 3]
```diff
@@ -235,15 +233,21 @@ impl VM {
...
-            .skip(code_offset)
-            .take(size)
-            .enumerate()
-        {
-            if let Some(data_byte) = data.get_mut(i) {
-                *data_byte = *byte;
+        if code_offset < current_call_frame.bytecode.len().into() {
+            let code_offset: usize = code_offset
+                .try_into()
+                .map_err(|_| VMError::VeryLargeNumber)?;
```
Comment (JereSalo) [inline crates/vm/levm/src/opcode_handlers/environment.rs:239]: Maybe this should be an internal error because it shouldn't ever fail. Right?

**Review by JereSalo — CHANGES_REQUESTED** [review]
The PR is good, maybe we should change `VeryLargeNumber` for an internal error (`ConversionError`) because we shouldn't revert execution if that fails at that point. That shouldn't ever happen.

File: crates/vm/levm/src/opcode_handlers/environment.rs (hunk 3 above)
Comment (lima-limon-inc) [reply crates/vm/levm/src/opcode_handlers/environment.rs]: Noted! Changed here: [374d904e9916f9b6559a6db3be259060893bc0f0](https://github.com/lambdaclass/ethrex/pull/1465/commits/374d904e9916f9b6559a6db3be259060893bc0f0)
Thanks Jere!

**Pull request merged**